### PR TITLE
new rebase of color-libraries

### PR DESCRIPTION
**Pull request by mkborregaard**
I copy-pasted the changes over manually

### CODE REVIEW

File: src/color_gradients.jl [hunk 1]
```diff
@@ -1,9 +1,74 @@
...
+    haskey(cl.lib, gradient) && return cl.lib[gradient]
+
+    potentials = [name for (name, library) in color_libraries if haskey(library.lib, gradient)]
+    length(potentials) == 0 && error("There is no gradient named $gradient . Use cgradients() to get a list of gradients in the current color library, clibraries() to get a list of available color libraries")
+    length(potentials) > 1 && warn("$gradient is found in more than one library: $(join(potentials, ", ")). Choosing $(potentials[1])")
+    color_libraries[potentials[1]][gradient]
+end
+
+getindex(cl::ColorLibrary, key::Symbol) = getgradient(key, cl)
+
```
Comment (mkborregaard) [inline src/color_gradients.jl]: Might be nice with a `setindex` function as well to easily add gradients (quite useful for PlotThemes, I'd imagine)

**Comment by tbreloff** [comment]
merge when you're comfortable!

**Comment by mkborregaard** [comment]
Thanks! I still only have push rights to StatPlots though :-)

**Comment by tbreloff** [comment]
Try again?

**Comment by mkborregaard** [comment]
Wohoo! :-)

**Comment by mkborregaard** [comment]
There are now PRs on PlotThemes and Plots that resolves the conflict with using themes.

**Pull request merged**